lib/IDE/CodeCompletion.cpp

```diff
@@ -5795,9 +5811,203 @@ static void addConditionalCompilationFlags(ASTContext &Ctx,
   }
 }
 
+static void deliverCompletionResults(CodeCompletionContext &CompletionContext,
+                                     CompletionLookup &Lookup,
+                                     SourceFile &SF,
+                                     CodeCompletionConsumer &Consumer) {
```

Comment:
I believe you just factored the code out from `CodeCompletionCallbacksImpl`. So i'm not reviewing the actual logic here. If you made any behavior changes, let me know.

Reply:
I think I may have changed everything to use the passed-in SF, while previously it was using P.SF in some places and CurDeclContext->getParentSourceFile() in others. Is that safe?